ocaml/xapi/vm_platform.ml

```diff
@@ -196,21 +196,17 @@ let sanity_check ~platformdata ?firmware ~vcpu_max ~vcpu_at_startup ~domain_type
         if cps < 1 || vcpus mod cps <> 0 then
           raise
             (Api_errors.Server_error
-               ( Api_errors.invalid_value
+               ( Api_errors.vcpu_max_not_cores_per_socket_multiple
                , [
-                   Printf.sprintf "platform:cores-per-socket (value %d)" cps
-                 ; Printf.sprintf
-                     "VCPUs_max (value %d) must be a multiple of \
-                      cores-per-socket"
-                     vcpus
-                 ] ))
+                   string_of_int vcpus; cps_str
+                 ]
+ ))
       with Failure msg ->
         raise
           (Api_errors.Server_error
              ( Api_errors.invalid_value
              , [
-                 Printf.sprintf "platform:cores-per-socket (value %s)" cps_str
-               ; "Value is not a valid int"
+                 Printf.sprintf "platform:cores-per-socket (value %s)" cps_str;
```

Comment:
I would leave it as it was, it was explaining the root of the problem :)

Reply:
The error is defined as:
```
  error Api_errors.invalid_value ["field"; "value"]
    ~doc:"The value given is invalid" ();
```
Therefore, it needs two arguments: the field name and the input value. No other text, just plain data.

So:
```
[ "platform:cores-per-socket"; cps_str]
```